Client/Frontend/UserContent/UserScripts/PaymentRequest.js

```diff
@@ -17,13 +21,31 @@ class $<paymentreq> {
   constructor (methodData, details) {
       this.methodData = JSON.stringify(methodData)
       this.details = JSON.stringify(details)
+      this.state = $<paymentreq>_CREATED
   }
 
   canMakePayment() {
-    return true;
+    const methodData = JSON.parse(this.methodData)
+    const details = JSON.parse(this.details)
+    const state = this.state
+    return new Promise(
+      function (resolve, reject) {
+        if (state != $<paymentreq>_CREATED) {
+          reject(new DOMException("canMakePayment error", "InvalidStateError"))
+          return
+        }
+        if (!methodData.some(e => e.supportedMethods === 'bat')) {
```

Comment:
Is it possible this will ever be "BAT"/"Bat"/etc. Should we normalize the check so that its always checking against lowercase or uppercase?

Reply:
All other instruments are lowercased, we have it `bat` in the reference code as well: https://github.com/brave-experiments/sku-js-php/blob/master/lib/js/sku.js#L159